webapp/channels/src/components/admin_console/team_channel_settings/team/details/index.ts

```diff
@@ -17,16 +17,19 @@ import {
 } from 'mattermost-redux/actions/groups';
 
 import {setNavigationBlocked} from 'actions/admin_actions';
-
-import TeamDetails from './team_details';
-
-function mapStateToProps(state, props) {
-    const teamID = props.match.params.team_id;
-    const team = getTeam(state, teamID);
-    const groups = getGroupsAssociatedToTeam(state, teamID);
-    const allGroups = getAllGroups(state, teamID);
-    const totalGroups = groups.length;
-    const isLicensedForLDAPGroups = state.entities.general.license.LDAPGroups === 'true';
+import TeamDetails, {Props as TeamDetailsProps} from './team_details';
+import {GlobalState} from 'types/store';
+import {Team} from '@mattermost/types/teams';
+import {Group} from '@mattermost/types/groups';
+import {ActionFunc, GenericAction} from 'mattermost-redux/types/actions';
+
+function mapStateToProps(state: GlobalState, props: TeamDetailsProps) {
+    const teamID: string = props.teamID;
+    const team: Team = getTeam(state, teamID);
+    const groups: Group[] = getGroupsAssociatedToTeam(state, teamID);
+    const allGroups: Record<string, Group> = getAllGroups(state);
+    const totalGroups: number = groups.length;
+    const isLicensedForLDAPGroups: boolean = state.entities.general.license.LDAPGroups === 'true';
```

Comment:
lets remove this explicit typing since they the types are returned from the selectors themselves.